qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -193,60 +204,164 @@ def _default_options(self):
             meas_return="single",
         )
 
-    def _draw_iq_shots(self, prob, shots, phase: float = 0.0) -> List[List[List[float]]]:
-        """Produce an IQ shot."""
-
-        rand_i = self._rng.normal(0, self._iq_cluster_width, size=shots)
-        rand_q = self._rng.normal(0, self._iq_cluster_width, size=shots)
-
-        memory = []
-        for idx, state in enumerate(self._rng.binomial(1, prob, size=shots)):
+    @staticmethod
+    def _verify_parameters(output_length: int, prob_dict: Dict[str, float]):
+        if output_length < 1:
+            raise ValueError(f"The output length {output_length} is smaller than 1.")
+
+        if not np.allclose(1, sum(prob_dict.values())):
+            raise ValueError("The probabilities given don't sum up to 1.")
+        for key in prob_dict.keys():
+            if output_length is not len(key):
+                raise ValueError(
+                    "The output lengths of the circuit and the output lengths in the dictionary"
+                    " don't match."
+                )
+
+    def _get_normal_samples_for_shot(self, num_qubits: int) -> np.ndarray:
+        """
+        Produce a list in the size of num_qubits. Each entry value is produced from normal distribution
+        with expected value of '0' and standard deviation of self._iq_cluster_width.
+        Args:
+            num_qubits(int): The amount of qubits in the circuit.
```

Comment:
```suggestion
            num_qubits(int): The number of qubits in the circuit.
```
I think (not sure) that "amount" tends to refer to non-discrete quantities.

Reply:
I checked and you are right:
```
Number refers to how many of something there is: how many mice, how many mutations, or how many calcium channels, for example. In other words, number refers to items we can count. By contrast, amount refers to how much of something there is: how much stimulation, how much resistance, or how much liquid.
```
[Link](http://crosstalk.cell.com/blog/the-difference-between-number-and-amount#:~:text=Number%20refers%20to%20how%20many,resistance%2C%20or%20how%20much%20liquid.)

changed as suggested.